dhis-2/dhis-api/src/main/java/org/hisp/dhis/mapping/MapView.java

```diff
@@ -525,6 +532,20 @@ public void setLegendSet( LegendSet legendSet )
         this.legendSet = legendSet;
     }
 
+    @JsonProperty
+    @JsonSerialize( as = BaseIdentifiableObject.class )
+    @JacksonXmlProperty( namespace = DxfNamespaces.DXF_2_0 )
+    @PropertyRange( min = 7, max = 7 )
+    public String getNoDataColor()
```

Comment:
What happens if the size is outside the range? Do you get an exception on persistence?

Reply:
We have built-in validation in the metadata importer linked to the @PropertyRange annotation. So you will get an import conflict from the API endpoint.